app/src/main/java/com/android/sample/ui/authentication/SignIn.kt

```diff
@@ -55,43 +57,56 @@ fun SignInScreen(navigationActions: NavigationActions, viewModel: SignInViewMode
-                  painter = painterResource(id = R.drawable.aptivity_logo),
-                  contentDescription = "App Logo",
-                  modifier = Modifier.size(110.dp).testTag("AppLogo"))
-              Spacer(modifier = Modifier.height(48.dp))
-
-              // Email Input
-              EmailTextField(
-                  email = emailState.value,
-                  onEmailChange = { emailState.value = it },
-                  emailError = null)
-              Spacer(modifier = Modifier.height(16.dp))
-
-              TextFieldWithErrorState(
-                  value = passwordState.value,
-                  onValueChange = { passwordState.value = it },
-                  label = "Password",
-                  modifier = Modifier.fillMaxWidth(0.8f).testTag("PasswordTextField"),
-                  validation = { input ->
-                    if (input.isBlank()) "Password cannot be empty" else null
-                  },
-                  externalError = passwordErrorState.value,
-                  errorTestTag = "PasswordErrorText")
-
-              Spacer(modifier = Modifier.height(16.dp))
-
-              // Sign-In Button
-              Button(
-                  onClick = {
-                    if (passwordState.value.isEmpty()) {
+            verticalArrangement = Arrangement.Center,
+        ) {
+          item {
+            // App Logo
+            Image(
+                painter = painterResource(id = R.drawable.aptivity_logo),
+                contentDescription = "App Logo",
+                modifier = Modifier.size(110.dp).testTag("AppLogo"))
+            Spacer(modifier = Modifier.height(48.dp))
+          }
+
+          item {
+            // Email Input
+            EmailTextField(
+                email = emailState.value,
+                onEmailChange = {
+                  emailState.value = it
+                  emailErrorState.value =
+                      if (!isValidEmail(it)) "Please enter a valid address: example@mail.xx "
+                      else null
+                },
+                emailError = emailErrorState.value)
+            Spacer(modifier = Modifier.height(16.dp))
+          }
+
+          item {
+            PasswordTextField(
+                password = passwordState.value,
+                onPasswordChange = { passwordState.value = it },
+                isPasswordVisible = isPasswordVisible.value,
+                onPasswordVisibilityChange = { isPasswordVisible.value = !isPasswordVisible.value },
+                passwordError = passwordErrorState.value)
+            Spacer(modifier = Modifier.height(16.dp))
+          }
+
+          item {
+            // Sign-In Button
+            Button(
+                onClick = {
+                  when {
+                    !isValidEmail(emailState.value) -> {
+                      emailErrorState.value = "Please enter a valid address: example@mail.xx"
+                    }
+                    passwordState.value.isEmpty() -> {
```

Comment:
Nice use of LazyColumn here! Since LazyColumn only renders the items that are currently visible on the screen, this should help improve performance, especially if the list grows in size.